packages/truffle-compile/compilerSupplier/loadingStrategies/LoadingStrategy.js

```diff
@@ -15,11 +15,15 @@ class LoadingStrategy {
     };
     this.config = Object.assign({}, defaultConfig, options);
 
-    const compilerCachePath = path.resolve(
+    let compilerCachePath = path.resolve(
```

Comment:
This solution will not work. If we are going to change the cache path to `~/.config/truffle/compilers/node_modules` then it needs to be done here. You can't change it later because it will only run through that if statement when it sees the directory doesn't exist. So the subsequent times it will cache things once again at `~/.config/truffle/compilers` and not in the `node_modules` folder inside `compilers`.

Reply:
Ah, I see. The reviewed code won't work for 5.0.8 users who've been using BYOC.

Thanks for that. Hopefully we can remove some of this code at a later date and/or find a more long-term solution.